src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorResolver.cs

```diff
@@ -67,8 +74,7 @@ public IEnumerable<TagHelperDescriptor> Resolve([NotNull] TagHelperDescriptorRes
                 }
                 catch (Exception ex)
                 {
-                    var directiveName = "@" + directiveDescriptor.DirectiveType.ToString().ToLowerInvariant();
-
+                    var directiveName = "@" + _directiveNames[directiveDescriptor.DirectiveType];
```

Comment:
Since we're pulling from a dictionary here we should try and retrieve the directive name, if we fail polyfill with something like `unknown`.